File: mindy/week1/O-4375.py
# input처리를 위해 try, catch 사용
# 1,11,111,...늘려가며 n으로 나누어 떨어지는지 확인

import sys

input = sys.stdin.readline

while True:
    try:
        n = int(input().strip())
         
    except:    
        break
    num=1
    count = 1
    while True:
            if num %n==0:
                break
            else:
                num = num*10+1
                count +=1
                
    print(count)   
            

File: leo/week1,2/O-4375.js
/**
 *  제목 :
 *  입력 :
 *  제한사항 :
 *  시간계산 필요 :
 */

// Run by Node.js
const readline = require("readline");

const solve = (n) => {
  let num = 1n;
  let count = 1n;

  while (true) {
    if (num % n === 0n) {
      return count;
    }
    count += 1n;
    num = num * 10n + 1n; // 다음 1로만 이루어진 수
  }
};

(async () => {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  const results = [];

  for await (const line of rl) {
    const n = BigInt(line.trim());
    if (n === null || n === 0n) break;
    results.push(solve(n).toString());
  }

  console.log(results.join("\n"));

  rl.close();
  process.exit();
})();
